packages/react-dom/src/server/ReactPartialRendererHooks.js

```diff
@@ -196,7 +196,12 @@ export function useReducer<S, A>(
           // priority because it will always be the same as the current
           // render's.
           const action = update.action;
+
+          currentlyRenderingComponent = null;
+          // now, it'll throw if you try to call a hook inside the reducer
           newState = reducer(newState, action);
+          // restore the current component
+          currentlyRenderingComponent = cachedCurrentlyRenderingComponent;
           update = update.next;
```

Comment:
Nit: I would replace a comment structure like

```js
    currentlyRenderingComponent = null;
    // now, it'll throw if you try to call a hook inside the reducer
    newState = reducer(newState, action);
    // restore the current component
    currentlyRenderingComponent = cachedCurrentlyRenderingComponent;
```

with something like

```js
    // Temporarily clear to forbid calling Hooks.
    currentlyRenderingComponent = null;
    newState = reducer(newState, action);
    currentlyRenderingComponent = cachedCurrentlyRenderingComponent;
```

The pattern of replacing something and then setting again is very common in the codebase so we don't need to over-explain it — just enough to hint at the purpose when we clear.

Reply:
It also helps to avoid newlines in between so that the starting and ending line of replacement are as close to each other as possible.